vowpalwabbit/stagewise_poly.cc

```diff
@@ -75,7 +75,7 @@ struct stagewise_poly
     cout << "total feature number (after poly expansion!) = " << sum_sparsity << std::endl;
 #endif  // DEBUG
 
-    synth_ec.feature_space[tree_atomics].delete_v();
+    //synth_ec.feature_space[tree_atomics].delete_v();
```

Comment:
Memory leak?

Reply:
More generally, have you verified things are sound with valgrind?

Reply:
`struct features` calls delete_v on its v_arrays in the destructor. So once `synth_ec` is destroyed, the features array goes out of scope and cleans itself

Reply:
> More generally, have you verified things are sound with valgrind?

Only insofar as to ensure there's no invalid memory usage or double freeing. I didn't check closely for leaks

Reply:
Definitely clean up all leaks before merging.

Reply:
All the leaks found in RunTests have been cleaned up (other than the 5 test that we know will fail because of bash/python)